orchagent/flex_counter/flex_counter_manager.cpp

```diff
@@ -0,0 +1,203 @@
+}
+
+// disableFlexCounterGroup will do nothing if the flex counter group has been
+// disabled.
+void FlexCounterManager::disableFlexCounterGroup()
+{
+    SWSS_LOG_ENTER();
+
+    if (!enabled)
+    {
+        return;
+    }
+
+    vector<FieldValueTuple> field_values =
+    {
+        FieldValueTuple(FLEX_COUNTER_STATUS_FIELD, FLEX_COUNTER_DISABLE)
+    };
+    flex_counter_group_table->set(group_name, field_values);
+    enabled = false;
+
+    SWSS_LOG_DEBUG("Disabling flex counters for group '%s'.", 
+            group_name.c_str());
+}
+
+// setCounterIdList configures a flex counter to poll the set of provided stats
+// that are associated with the given object.
+void FlexCounterManager::setCounterIdList(
+        const sai_object_id_t object_id,
+        const CounterType counter_type,
+        const unordered_set<string> &counter_stats)
+{
+    SWSS_LOG_ENTER();
+
+    auto counter_type_it = counter_id_field_lookup.find(counter_type);
+    if (counter_type_it == counter_id_field_lookup.end())
+    {
+        SWSS_LOG_ERROR("Could not update flex counter id list for group '%s': counter type not found.",
+                group_name.c_str());
+        return;
+    }
+
+    std::vector<swss::FieldValueTuple> field_values =
+    {
+        FieldValueTuple(counter_type_it->second, serializeCounterStats(counter_stats))
+    };
+    flex_counter_table->set(
+            getFlexCounterTableKey(group_name, object_id), field_values);
+    installed_counters.insert(object_id);
+
+    SWSS_LOG_DEBUG("Updated flex counter id list for object '%lu' in group '%s'.", 
+            object_id,
+            group_name.c_str());
+}
+
+// clearCounterIdList clears all stats that are currently being polled from 
+// the given object.
+void FlexCounterManager::clearCounterIdList(const sai_object_id_t object_id)
+{
+    SWSS_LOG_ENTER();
+
+    auto counter_it = installed_counters.find(object_id);
+    if (counter_it == installed_counters.end())
+    {
+        SWSS_LOG_DEBUG("No counters found on object '%lu' in group '%s'.",
+                object_id,
+                group_name.c_str());
+        return;
+    }
+
+    flex_counter_table->del(getFlexCounterTableKey(group_name, object_id));
+    installed_counters.erase(counter_it);
+
+    SWSS_LOG_DEBUG("Cleared flex counter id list for object '%lu' in group '%s'.",
+            object_id,
+            group_name.c_str());
+}
+
+string FlexCounterManager::getFlexCounterTableKey(
+        const string &group_name, 
+        const sai_object_id_t object_id) const
+{
+    SWSS_LOG_ENTER();
+
+    return group_name + ':' + sai_serialize_object_id(object_id);
+}
+
+// serializeCounterStats turns a set of stats into a format suitable for FLEX_COUNTER_DB.
+string FlexCounterManager::serializeCounterStats(
+        const unordered_set<string> &counter_stats) const
+{
+    SWSS_LOG_ENTER();
+
+    string stats_string;
+    for (const auto &stat : counter_stats) 
+    {
+        stats_string.append(",");
```

Comment:
>stats_string.append(","); [](start = 8, length = 25)

Don't append before first stat.

Reply:
This is actually intentional but may have been unclear. I've added a comment to clarify the purpose.

Reply:
I know your intention. I mean don't append before first component, not a good practice.